### PR TITLE
Using standard layernorm cuda kernel for skiplayernorm.

### PR DESCRIPTION
**Pull request by zhanghuanrong**
* Current SkipLayernorm did not using stable algo and cause correctness issue.
  * Enrich existing layernorm kernel to accept bias and residual.
  * Tune standard layernorm threads.y according to elements and device property.
  * Remove existing skiplayernorm cuda implementation.

### CODE REVIEW

**Comment by tianleiwu** [comment]
Please run performance and accuracy tests on BERT and GPT-2 model to see the impact.

File: onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc [hunk 1]
```diff
@@ -106,25 +106,29 @@ Status SkipLayerNorm<T, Simplified>::ComputeInternal(OpKernelContext* ctx) const
...
-  int64_t element_count = input_dims[0] * sequence_length * hidden_size;
-  size_t element_size = sizeof(T);
-  typedef typename ToCudaType<T>::MappedType CudaT;
+  int sequence_length = gsl::narrow_cast<int>(input_dims[1]);
+  int hidden_size = gsl::narrow_cast<int>(input_dims[2]);
+  int row_count = gsl::narrow_cast<int>(input_dims[0] * sequence_length);
 
-  return LaunchSkipLayerNormKernel<CudaT, Simplified>(
+  typedef typename ToCudaType<T>::MappedType CudaT;
+  HostApplyLayerNorm<CudaT, float, CudaT, Simplified>(
```
Comment (hariharans29) [inline onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc:114]: Maybe we should just limit this change to GPT2-style SLN with a TODO to explore moving over altogether from current SLN implementation. Not sure what the accuracy and latency implications are for other models that have historically been using the "original" SLN implementation (for example BERT) ?

File: onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu [hunk 2]
```diff
@@ -1,241 +0,0 @@
...
-__global__ void SkipLayerNormKernel(
-    const int ld, const T* input, const T* skip,
-    const T* beta, const T* gamma, const T* bias,
-    const T epsilon, T* output, T* skip_input_bias_add_output) {
-  const T reverse_ld = T(1.f / ld);
-  const int offset = blockIdx.x * ld;
-
-  KeyValuePairSum pair_sum;
-  // reduce x and x^2
-  cub::KeyValuePair<T, T> thread_data(0, 0);
```
Comment (hariharans29) [inline onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu:91]: Should we check with the training team on whether it is okay to switch over completely from the existing implementation ? I think the training team introduced some optimizations in this file recently and they might still want to use this code path

Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu:91]: Add them to PR

File: onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc (hunk 1 above)
Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc:114]: Hope future optimization should on the Single LayerNorml Kernel.

**Review by yufenglee — APPROVED** [review]
:shipit:

**Pull request merged**